contracts/impls/FlowMarginProtocol2.sol

```diff
@@ -0,0 +1,180 @@
+pragma solidity ^0.6.3;
+
+import "@openzeppelin/upgrades/contracts/Initializable.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "@openzeppelin/contracts/math/SignedSafeMath.sol";
+import "@openzeppelin/contracts/math/Math.sol";
+
+import "../libs/Percentage.sol";
+import "../libs/upgrades/UpgradeOwnable.sol";
+import "../libs/upgrades/UpgradeReentrancyGuard.sol";
+
+import "../interfaces/PriceOracleInterface.sol";
+import "../interfaces/MoneyMarketInterface.sol";
+import "../interfaces/LiquidityPoolInterface.sol";
+import "./FlowToken.sol";
+
+contract Leverage {
+    function isLong() public pure returns(bool) {
```

Comment:
should just use `int256` to represent Leverage and some helper methods to decode

In Rust we make it an enum so we can pack it and fit everything in u16, but given u16 doesn't save spaces in solidity, just int256 will make things simpler.

The format can be:

256 bits, low 128 bits means long, high 128 bits means shorts

e.g. long 2 is 1 << 2, long 100 is 1 << 100
short 2 is 1 << (128 + 2) and short 100 is 1 << (128 + 100)

long 2 & short 2 is (1 << 2 | 1 << (128 + 2))

Reply:
Got it, I haven't spend much time thinking about the Leverage yet, just a dummy at the moment.